### PR TITLE
Add integer quantifiers

### LINKED ISSUE #335: Add Simple Quantifiers
At a minimum, should include:
    - IntegerLessThan
    - IntegerRange

### PR DESCRIPTION
**Pull request by karlfloersch**
## Description
This PR adds simple integer quantifiers `IntegerRangeQuantifier` and `NonnegativeIntegerQuantifier`.

## Questions
- Is it worth it to have a `NonnegativeIntegerQuantifier`? 
- Is it worth it to have entire "Quantifier" class for this suuuper simple quantifier?
- Why does this PR feel so trivial? 😄 

### Fixes
- Fixes #335 

## Contributing Agreement

- [x] I have read and understood the [Plasma Group Contributing Guide and Code of Conduct](https://github.com/plasma-group/pigi/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: packages/core/src/app/ovm/integer-quantifiers.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { QuantifierResult, Quantifier } from '../../types'
+
+// Helper function which returns an array of numbers, starting at start, ending at end, incrementing by 1.
+// Eg. [0, 1, 2,...end]
+const range = (start: number, end: number) =>
```
Comment (willmeister) [inline packages/core/src/app/ovm/integer-quantifiers.ts]: It's pretty obvious, but might suggest adding return type `(...) : number[] => ...`

File: packages/core/src/app/ovm/integer-quantifiers.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/*
+ * The parameter type for `getAllQuantified(...)` in the IntegerRangeQuantifier.
+ */
+interface IntegerRangeParameters {
+  start: number
+  end: number
+}
+
+/*
+ * The IntegerRangeQuantifier returns a range of integers between the start & end.
```
Comment (willmeister) [inline packages/core/src/app/ovm/integer-quantifiers.ts]: Change to "between the start (inclusive) & end (exclusive)"

File: packages/core/src/app/ovm/integer-quantifiers.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+export class IntegerRangeQuantifier implements Quantifier {
+  /**
+   * Returns a QuantifierResult where results are an array of integers from 0 to _lessThanThis. Eg. 0 to 3 = [0, 1, 2]
+   * and `allResultsQuantified` is set to `true`--this is because we always can quantify integers in this range.
+   *
+   * @param _withinThisRange the range of the integers we would like to return.
+   */
+  public getAllQuantified(_withinThisRange: {
```
Comment (willmeister) [inline packages/core/src/app/ovm/integer-quantifiers.ts]: nitpick, but I think parameters elsewhere don't have leading underscores.

File: packages/core/src/app/ovm/integer-quantifiers.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  start: number
+  end: number
+}
+
+/*
+ * The IntegerRangeQuantifier returns a range of integers between the start & end.
+ */
+export class IntegerRangeQuantifier implements Quantifier {
+  /**
+   * Returns a QuantifierResult where results are an array of integers from 0 to _lessThanThis. Eg. 0 to 3 = [0, 1, 2]
```
Comment (willmeister) [inline packages/core/src/app/ovm/integer-quantifiers.ts]: I think the comments here are copy/pasted from `NonnegativeIntegerLessThanQuantifier`.

File: packages/core/test/app/ovm/integer-quantifiers.spec.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/* Internal Imports */
+import {
+  IntegerRangeQuantifier,
+  NonnegativeIntegerLessThanQuantifier,
+} from '../../../src/app'
+
+describe('IntegerQuantifiers', () => {
+  describe('IntegerRangeQuantifier', () => {
```
Comment (willmeister) [inline packages/core/test/app/ovm/integer-quantifiers.spec.ts]: Would also test where `start === end`, when both are negative, when just one is negative, when `end < start`, etc.

File: packages/core/test/app/ovm/integer-quantifiers.spec.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      const quantifier = new IntegerRangeQuantifier()
+      const range = quantifier.getAllQuantified({ start: 100, end: 105 })
+      range.should.deep.equal({
+        results: [100, 101, 102, 103, 104],
+        allResultsQuantified: true,
+      })
+    })
+  })
+  describe('NonnegativeIntegerLessThanQuantifier', () => {
+    it('should quantify a range', async () => {
```
Comment (willmeister) [inline packages/core/test/app/ovm/integer-quantifiers.spec.ts]: Would test negative as well

File: packages/core/src/app/ovm/integer-quantifiers.ts (hunk 1 above)
Comment (karlfloersch) [reply packages/core/src/app/ovm/integer-quantifiers.ts]: Whoops!
